board/linux/README.md

```diff
@@ -0,0 +1,110 @@
+{
+    __dirstream *ret{};
+    if (!dirname) {
+        //...
+        errno = EINVAL;
+        return ret;
+    }
+    if (vfs::redirect_to_image(dirname)) {
+        //...
+    }
+    else {
+        char tmp[PATH_MAX];
+        // Path transtation
+        const auto newpath = vfs::npath_translate(dirname, tmp);
+        // Invoke opendir from glibc
+        ret = real::opendir(newpath);
+    }
+    return ret;
+}
+__asm__(".symver _iosys_opendir,opendir@GLIBC_2.2.5");
+```
+As you see it is done by `.symver _iosys_opendir,opendir@GLIBC_2.2.5` which links it with a proper version of `glibc` function.
+
+```
+        +--------------+                      +--------------------------+                  +-----------------+
+        |              |    invoke wrapper    | _iosys_opendir(...) {    |   invoke glibc   |                 |
+        | opendir(...) |--------------------->| opendir(translated_path) |----------------->| real::opendir() |
+        |              |                      | }                        |                  |                 |
+        +--------------+                      +--------------------------+                  +-----------------+
+```
+
+## How to find correct version
+Our library does not handle all syscalls because we don't know the exact syscall version. While changing compiler or some library
+it will be necessary to update the version or add a new wrapper. 
+
+Example:
+Let's say we have a new compiler and there is `stdc++` implementation that uses a different `stat` syscall or different version (see Related links).
+We can easily find the correct version by checking symbols.
+```
+$ readelf -Ws /usr/lib/x86_64-linux-gnu/libstdc++.so.6 | grep -w stat
+   176: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND stat@GLIBC_2.33 (66)
+```
+## How to update syscalls wrappers
+### Update version.txt
+This file is used by linker to tell which internal names to replace. Before editing check `glibc` version and do not duplicate symbols.
+```
+GLIBC_2.33 {
+    global:
+        stat;   # updated stat version inserting to GLIBC_2.33
+        lstat;
+        fstat;
+        stat64;
+        lstat64;
+        fstat64;
+};
+```
+### Add or update wrapper
+Updating the wrapper version is easy. You need to modify only `@GLIBC` string according to symbols found previously.
+```
+__asm__(".symver _iosys_stat,stat@GLIBC_<version>");
+```
+However if you need to add a new wrapper the best way is just to check others implementations. But finally it should look like:
+```
+_iosys_<function_name>(...)
+{
+    // VFS implementation if needed
+    if (...) {
+        //...
+    }
+    else {
+        // Path transtation if needed
+        // Invoke function_name from glibc id needed
+    }
+}
+__asm__(".symver _iosys_<function_name>,function_name@GLIBC_<version>");
+```
+### Add declarations (if needed)
+If you added wrapper you must add helpers for intercepting library calls:
+- `__REAL_DECL(function_name)` - just `decltype`
```

Comment:
Is is possible to explain more for these 2 lines?

Reply:
Done. I also added related link about that :)